autoscaler/kubernetes_client.go

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package autoscaler
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"k8s.io/client-go/1.4/kubernetes"
+	"k8s.io/client-go/1.4/pkg/api"
+	"k8s.io/client-go/1.4/pkg/api/resource"
+	apiv1 "k8s.io/client-go/1.4/pkg/api/v1"
+	"k8s.io/client-go/1.4/rest"
+)
+
+// K8sClientInterface - Wraps all needed client functionalities for autoscaler
+type K8sClientInterface interface {
+	GetScaler() (ScalerInterface, error)
+	FetchConfigMap(namespace, configmap, key string) (configMap ConfigMap, err error)
+	GetClusterStatus() (clusterStatus ClusterStatus, err error)
+	GetNamespace() (namespace string)
+}
+
+// KubernetesClient - Wraps all Kubernetes API client functionalities
+type KubernetesClient struct {
+	namespace string
+	target    ScaleTarget
+	clientset *kubernetes.Clientset
+}
+
+type ClusterStatus struct {
+	totalNodes       int32
+	schedulableNodes int32
+	totalCores       int32
+	schedulableCores int32
+}
+
+type ConfigMap struct {
+	data    string
+	version string
+}
+
+// ScaleTarget stores the scalable target recourse
+type ScaleTarget struct {
+	targetType string
+	targetName string
+}
+
+// GetNamespace returns the namespace of current k8s client.
+func (k *KubernetesClient) GetNamespace() (namespace string) {
+	return k.namespace
+}
+
+// GetScaler returns a ScalerInterface for client's target resource.
+func (k *KubernetesClient) GetScaler() (ScalerInterface, error) {
+	switch k.target.targetType {
+	case "deployment":
```

Comment:
Checked the main repo but didn't find any obvious place to import this. And because this `targetType` is only used here and when the user inputs target argument, we won't need to change this as long as we don't modify the input pattern.

We may do this in a better way in the future but I want to defer this for now.